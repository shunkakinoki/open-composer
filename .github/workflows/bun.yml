name: Bun
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
  workflow_dispatch:
concurrency:
  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.head_ref || github.sha }}
  cancel-in-progress: true
jobs:
  bun-build:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Build
        run: bun run build
  bun-lint:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Check
        run: bun run check
      - name: Format
        run: bun run format
      - name: Lint
        run: bun run lint
      - name: Type Check
        run: bun run type-check
  bun-test:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Test
        run: bun run test
  bun-check:
    if: always()
    needs:
      - bun-build
      - bun-lint
      - bun-test
    runs-on: ubuntu-latest
    timeout-minutes: 3
    steps:
      - name: Alls Green
        uses: re-actors/alls-green@release/v1
        with:
          jobs: ${{ toJSON(needs) }}
