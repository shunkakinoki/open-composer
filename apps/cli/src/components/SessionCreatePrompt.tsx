import { DatabaseLive } from "@open-composer/db";
import * as Effect from "effect/Effect";
import { Box, Text, useApp, useInput } from "ink";
import type React from "react";
import { useState } from "react";
import { SessionsService } from "../services/sessions-service.js";

/**
 * Interactive React component for creating development sessions.
 *
 * Only prompts for information that wasn't provided via props:
 * - If initialName is not provided, prompts for session name
 * - If initialWorkspaceChoice is not provided, prompts for workspace choice
 * - If workspace choice requires a path and initialWorkspacePath is not provided, prompts for path
 * - Otherwise, shows confirmation and allows immediate creation
 *
 * @example
 * // Prompt for everything
 * <SessionCreatePrompt onComplete={(id) => console.log(`Created session ${id}`)} />
 *
 * @example
 * // Only prompt for workspace choice and path
 * <SessionCreatePrompt
 *   initialName="My Project"
 *   onComplete={(id) => console.log(`Created session ${id}`)}
 * />
 *
 * @example
 * // Only prompt for workspace path
 * <SessionCreatePrompt
 *   initialName="My Project"
 *   initialWorkspaceChoice="existing"
 *   onComplete={(id) => console.log(`Created session ${id}`)}
 * />
 *
 * @example
 * // No prompts - direct confirmation
 * <SessionCreatePrompt
 *   initialName="My Project"
 *   initialWorkspaceChoice="none"
 *   onComplete={(id) => console.log(`Created session ${id}`)}
 * />
 */

type WorkspaceChoice = "existing" | "create" | "none";
type Step = "name" | "workspace-choice" | "workspace-path" | "confirm";

interface SessionCreatePromptProps {
  initialName?: string;
  initialWorkspaceChoice?: "existing" | "create" | "none";
  initialWorkspacePath?: string;
  onComplete: (sessionId: number) => void;
  onCancel?: () => void;
}

export const SessionCreatePrompt: React.FC<SessionCreatePromptProps> = ({
  initialName,
  initialWorkspaceChoice,
  initialWorkspacePath,
  onComplete,
  onCancel,
}) => {
  // Determine initial step based on provided arguments
  const getInitialStep = (): Step => {
    if (!initialName) return "name";
    if (!initialWorkspaceChoice) return "workspace-choice";
    if (
      (initialWorkspaceChoice === "existing" ||
        initialWorkspaceChoice === "create") &&
      !initialWorkspacePath
    )
      return "workspace-path";
    return "confirm";
  };

  const [step, setStep] = useState<Step>(getInitialStep());
  const [sessionName, setSessionName] = useState(initialName || "");
  const [workspaceChoice, setWorkspaceChoice] = useState<WorkspaceChoice>(
    initialWorkspaceChoice || "existing",
  );
  const [workspacePath, setWorkspacePath] = useState(
    initialWorkspacePath || "",
  );
  const [isCreating, setIsCreating] = useState(false);
  const [error, setError] = useState<string | null>(null);
  const { exit } = useApp();

  const handleCreateSession = async () => {
    setIsCreating(true);
    setError(null);

    try {
      const cli = new SessionsService();
      const sessionId = await cli
        .createInteractive(sessionName, workspaceChoice, workspacePath)
        .pipe(Effect.provide(DatabaseLive), Effect.runPromise);
      onComplete(sessionId);
      exit();
    } catch (err) {
      setError(err instanceof Error ? err.message : "Failed to create session");
      setIsCreating(false);
    }
  };

  useInput(
    (input, key) => {
      if (isCreating) return;

      if (key.escape || (key.ctrl && input === "c")) {
        onCancel?.();
        exit();
        return;
      }

      switch (step) {
        case "name":
          if (key.return) {
            if (sessionName.trim()) {
              setStep("workspace-choice");
            } else {
              setSessionName(`Session ${Date.now()}`);
              setStep("workspace-choice");
            }
          } else if (key.backspace || key.delete) {
            setSessionName(sessionName.slice(0, -1));
          } else if (input && !key.ctrl && !key.meta) {
            setSessionName(sessionName + input);
          }
          break;

        case "workspace-choice":
          if (key.upArrow || key.downArrow) {
            const choices: WorkspaceChoice[] = ["existing", "create", "none"];
            const currentIndex = choices.indexOf(workspaceChoice);
            const nextIndex = key.upArrow
              ? (currentIndex - 1 + choices.length) % choices.length
              : (currentIndex + 1) % choices.length;
            setWorkspaceChoice(choices[nextIndex]);
          } else if (key.return) {
            if (workspaceChoice === "none") {
              setStep("confirm");
            } else {
              setStep("workspace-path");
            }
          }
          break;

        case "workspace-path":
          if (key.return) {
            if (workspacePath.trim()) {
              setStep("confirm");
            }
          } else if (key.backspace || key.delete) {
            setWorkspacePath(workspacePath.slice(0, -1));
          } else if (input && !key.ctrl && !key.meta) {
            setWorkspacePath(workspacePath + input);
          }
          break;

        case "confirm":
          if (key.return) {
            handleCreateSession();
          }
          break;
      }
    },
    { isActive: true },
  );

  const renderStep = () => {
    switch (step) {
      case "name":
        return (
          <Box flexDirection="column">
            <Text bold color="cyan">
              üéØ Create New Session
            </Text>
            <Box marginTop={1}>
              <Text>Enter session name (press Enter for auto-generated):</Text>
            </Box>
            <Box marginTop={1}>
              <Text color="green">{sessionName}</Text>
              <Text color="gray">_</Text>
            </Box>
            <Box marginTop={1}>
              <Text color="gray">Press Enter to continue, Esc to cancel</Text>
            </Box>
          </Box>
        );

      case "workspace-choice":
        return (
          <Box flexDirection="column">
            <Text bold color="cyan">
              üìÅ Choose Workspace Option
            </Text>
            <Box marginTop={1}>
              <Text>Choose workspace option:</Text>
            </Box>
            <Box marginTop={1} flexDirection="column">
              <Box>
                <Text color={workspaceChoice === "existing" ? "green" : "gray"}>
                  {workspaceChoice === "existing" ? "‚óè" : "‚óã"} 1. Use existing
                  git workspace
                </Text>
              </Box>
              <Box marginTop={1}>
                <Text color={workspaceChoice === "create" ? "green" : "gray"}>
                  {workspaceChoice === "create" ? "‚óè" : "‚óã"} 2. Create new
                  workspace
                </Text>
              </Box>
              <Box marginTop={1}>
                <Text color={workspaceChoice === "none" ? "green" : "gray"}>
                  {workspaceChoice === "none" ? "‚óè" : "‚óã"} 3. No workspace (just
                  session tracking)
                </Text>
              </Box>
            </Box>
            <Box marginTop={2}>
              <Text color="gray">
                Use ‚Üë‚Üì to select, Enter to confirm, Esc to cancel
              </Text>
            </Box>
          </Box>
        );

      case "workspace-path": {
        const placeholder =
          workspaceChoice === "existing"
            ? "Enter path to existing git workspace"
            : "Enter path for new workspace";

        return (
          <Box flexDirection="column">
            <Text bold color="cyan">
              üìÇ{" "}
              {workspaceChoice === "existing"
                ? "Select Existing"
                : "Create New"}{" "}
              Workspace
            </Text>
            <Box marginTop={1}>
              <Text>{placeholder}:</Text>
            </Box>
            <Box marginTop={1}>
              <Text color="yellow">{workspacePath}</Text>
              <Text color="gray">_</Text>
            </Box>
            <Box marginTop={1}>
              <Text color="gray">Press Enter to continue, Esc to cancel</Text>
            </Box>
          </Box>
        );
      }

      case "confirm":
        return (
          <Box flexDirection="column">
            <Text bold color="cyan">
              ‚úÖ Confirm Session Creation
            </Text>
            <Box marginTop={1}>
              <Text>
                Session Name: <Text color="green">{sessionName}</Text>
              </Text>
            </Box>
            <Box marginTop={1}>
              <Text>Workspace: </Text>
              {workspaceChoice === "none" ? (
                <Text color="gray">No workspace assigned</Text>
              ) : (
                <Text color="yellow">{workspacePath}</Text>
              )}
            </Box>
            <Box marginTop={2}>
              <Text color="gray">
                Press Enter to create session, Esc to cancel
              </Text>
            </Box>
            {error && (
              <Box marginTop={1}>
                <Text color="red">‚ùå {error}</Text>
              </Box>
            )}
          </Box>
        );
    }
  };

  if (isCreating) {
    return (
      <Box flexDirection="column" padding={2}>
        <Text bold color="cyan">
          üîÑ Creating Session...
        </Text>
        <Box marginTop={1}>
          <Text>Please wait while we set up your session.</Text>
        </Box>
      </Box>
    );
  }

  return (
    <Box flexDirection="column" padding={2}>
      {renderStep()}
    </Box>
  );
};
