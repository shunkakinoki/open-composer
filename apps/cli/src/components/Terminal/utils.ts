/**
 * Terminal utility functions
 * Inspired by termact: https://github.com/MasterGordon/termact/blob/main/src/utils.ts
 */

import type { TerminalStyle } from "./types.js";
import {
  bg,
  blink,
  bold,
  doubleUnderline,
  fg,
  hidden,
  inverse,
  italic,
  strikethrough,
  underline,
} from "./ansi.js";

export const applyStyle = (style: TerminalStyle): string => {
  let result = "";
  if (style.fg) result += fg(style.fg);
  if (style.bg) result += bg(style.bg);
  if (style.bold) result += bold();
  if (style.italic) result += italic();
  if (style.underline) result += underline();
  if (style.blink) result += blink();
  if (style.reverse) result += inverse();
  if (style.hidden) result += hidden();
  if (style.strikethrough) result += strikethrough();
  if (style.doubleUnderline) result += doubleUnderline();
  return result;
};

export const wrapText = (text: string, width: number): string[] => {
  const lines: string[] = [];
  const words = text.split(" ");
  let currentLine = "";

  for (const word of words) {
    if (currentLine.length + word.length + 1 <= width) {
      currentLine += (currentLine ? " " : "") + word;
    } else {
      if (currentLine) lines.push(currentLine);
      currentLine = word;
    }
  }

  if (currentLine) lines.push(currentLine);
  return lines;
};

export const truncateText = (text: string, maxLength: number): string => {
  if (text.length <= maxLength) {
    return text;
  }
  return text.slice(0, maxLength - 3) + "...";
};

export const padText = (
  text: string,
  width: number,
  align: "left" | "center" | "right" = "left",
): string => {
  if (text.length >= width) {
    return text.slice(0, width);
  }

  const padding = width - text.length;

  switch (align) {
    case "center": {
      const leftPad = Math.floor(padding / 2);
      const rightPad = padding - leftPad;
      return " ".repeat(leftPad) + text + " ".repeat(rightPad);
    }
    case "right":
      return " ".repeat(padding) + text;
    default:
      return text + " ".repeat(padding);
  }
};

export const getNtscGrayscale = (r: number, g: number, b: number): number => {
  return r * 0.299 + g * 0.587 + b * 0.114;
};
