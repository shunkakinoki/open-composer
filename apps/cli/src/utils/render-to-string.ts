import { render as inkRender } from "ink";
import type { ReactElement } from "react";
import { EventEmitter } from "node:events";

/**
 * Minimal writable stream for capturing Ink output
 */
class CaptureStream extends EventEmitter {
  private output = "";

  write(chunk: string): void {
    this.output += chunk;
  }

  getOutput(): string {
    return this.output;
  }

  end(): void {
    // no-op
  }
}

/**
 * Renders a React component to a string using Ink
 * Useful for rendering UI components in Effect CLI commands
 */
export function renderToString(element: ReactElement): string {
  const stdout = new CaptureStream();
  const stderr = new CaptureStream();

  const instance = inkRender(element, {
    stdout: stdout as unknown as NodeJS.WriteStream,
    stderr: stderr as unknown as NodeJS.WriteStream,
    debug: false,
    exitOnCtrlC: false,
    patchConsole: false,
  });

  // Wait for a tick to let Ink render
  const output = stdout.getOutput();

  // Clean up
  instance.unmount();
  instance.cleanup();

  return output;
}
