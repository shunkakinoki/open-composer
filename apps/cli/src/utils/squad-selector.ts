import inquirer from "inquirer";
import type { LLMTask } from "@open-composer/agent-registry";
import { getAgentRegistry, getSquadLauncher } from "@open-composer/agent-registry";

/**
 * Start interactive squad selector
 */
export async function startSquadSelector(): Promise<void> {
  const registry = getAgentRegistry();
  const launcher = getSquadLauncher();

  console.log("\n🎮 Pokemon-Style Squad Manager\n");

  const { action } = await inquirer.prompt([
    {
      type: "list",
      name: "action",
      message: "What would you like to do?",
      choices: [
        { name: "📋 List all agents", value: "list" },
        { name: "✨ Create new squad", value: "create" },
        { name: "🚀 Quick squad (by task)", value: "quick" },
        { name: "📊 View statistics", value: "stats" },
        { name: "🚪 Exit", value: "exit" },
      ],
    },
  ]);

  switch (action) {
    case "list": {
      const agents = registry.getAllAgents();
      console.log(`\n📦 Found ${agents.length} agents:\n`);
      for (const agent of agents) {
        console.log(
          `  ${agent.pokemonAttributes.sprite} ${agent.name} (${agent.tier}) - ${agent.provider}`,
        );
      }
      break;
    }

    case "create": {
      console.log("\n🛠️  Create Squad - Not implemented in CLI yet");
      console.log("   Use the interactive UI components for full squad creation");
      break;
    }

    case "quick": {
      const { task } = await inquirer.prompt([
        {
          type: "list",
          name: "task",
          message: "Select task specialization:",
          choices: [
            { name: "🧭 Planner", value: "planner" },
            { name: "⚡ Programmer", value: "programmer" },
            { name: "🔍 Reviewer", value: "reviewer" },
            { name: "🌊 Router", value: "router" },
            { name: "🌿 Summarizer", value: "summarizer" },
          ],
        },
      ]);

      const team = launcher.createQuickSquad(task as LLMTask);
      console.log(`\n✅ Created squad: ${team.name}`);
      console.log(`   Agents: ${team.agents.map((a) => a.name).join(", ")}`);
      break;
    }

    case "stats": {
      const stats = registry.getStats();
      console.log("\n📊 Registry Statistics:");
      console.log(`   Total Agents: ${stats.totalAgents}`);
      console.log(`   Total Squads: ${stats.totalTeams}`);
      console.log(`   Starter: ${stats.tierCounts.starter || 0}`);
      console.log(`   Evolved: ${stats.tierCounts.evolved || 0}`);
      console.log(`   Legendary: ${stats.tierCounts.legendary || 0}`);
      break;
    }

    case "exit":
      console.log("\n👋 Goodbye!\n");
      return;
  }

  // Loop back to main menu
  await startSquadSelector();
}
