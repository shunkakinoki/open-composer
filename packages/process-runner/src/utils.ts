import * as Effect from "effect/Effect";
import {
  MAX_COMMAND_LENGTH,
  MAX_RUN_NAME_LENGTH,
  RUN_NAME_PATTERN,
} from "./constants.js";
import { ProcessRunnerError } from "./types.js";

// =============================================================================
// VALIDATION UTILITIES
// =============================================================================

export function validateRunName(
  runName: string,
): Effect.Effect<string, ProcessRunnerError> {
  if (!runName || typeof runName !== "string") {
    return Effect.fail(
      ProcessRunnerError("Run name must be a non-empty string"),
    );
  }

  if (runName.length > MAX_RUN_NAME_LENGTH) {
    return Effect.fail(
      ProcessRunnerError(
        `Run name too long (max ${MAX_RUN_NAME_LENGTH} characters)`,
      ),
    );
  }

  if (!RUN_NAME_PATTERN.test(runName)) {
    return Effect.fail(
      ProcessRunnerError(
        "Run name can only contain letters, numbers, hyphens, and underscores",
      ),
    );
  }

  return Effect.succeed(runName);
}

export function validateCommand(
  command: string,
): Effect.Effect<string, ProcessRunnerError> {
  if (!command || typeof command !== "string") {
    return Effect.fail(
      ProcessRunnerError("Command must be a non-empty string"),
    );
  }

  if (command.length > MAX_COMMAND_LENGTH) {
    return Effect.fail(
      ProcessRunnerError(
        `Command too long (max ${MAX_COMMAND_LENGTH} characters)`,
      ),
    );
  }

  // Basic command injection prevention - only reject null bytes
  if (command.includes("\x00")) {
    return Effect.fail(
      ProcessRunnerError("Command contains invalid characters"),
    );
  }

  return Effect.succeed(command);
}

// =============================================================================
// UTILITY FUNCTIONS
// =============================================================================

export function withTimeout<T>(
  effect: Effect.Effect<T, ProcessRunnerError>,
  timeoutMs: number,
  timeoutMessage: string = "Operation timed out",
): Effect.Effect<T, ProcessRunnerError> {
  const timeoutEffect = Effect.async<T, ProcessRunnerError>((resume) => {
    const timeout = setTimeout(() => {
      resume(Effect.fail(ProcessRunnerError(timeoutMessage)));
    }, timeoutMs);
    return Effect.sync(() => clearTimeout(timeout));
  });

  return Effect.race(effect, timeoutEffect);
}
