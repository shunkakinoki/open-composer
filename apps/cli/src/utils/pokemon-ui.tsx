import type { AgentTeam, OpenComposerAgent } from "@open-composer/agent-registry";
import type { RegistryStats } from "@open-composer/analytics";
import { renderToString } from "./render-to-string.js";
import {
  AgentCard,
  AgentComparison,
  AgentList,
  EvolutionTree,
  RegistryStats as RegistryStatsComponent,
  TaskSpecializations,
  TeamDisplay,
} from "../components/PokemonUI.js";

/**
 * Pokemon-style UI renderer using React components
 * This is a utility wrapper that renders React components to strings for CLI output
 */
export class PokemonUI {
  /**
   * Render agent card
   */
  static renderAgentCard(agent: OpenComposerAgent): string {
    return renderToString(<AgentCard agent={agent} />);
  }

  /**
   * Render agent list
   */
  static renderAgentList(agents: OpenComposerAgent[]): string {
    return renderToString(<AgentList agents={agents} />);
  }

  /**
   * Render team display
   */
  static renderTeam(team: AgentTeam): string {
    return renderToString(<TeamDisplay team={team} />);
  }

  /**
   * Render evolution tree
   */
  static renderEvolutionTree(agent: OpenComposerAgent): string {
    return renderToString(<EvolutionTree agent={agent} />);
  }

  /**
   * Render task specializations
   */
  static renderTaskSpecializations(agent: OpenComposerAgent): string {
    return renderToString(<TaskSpecializations agent={agent} />);
  }

  /**
   * Render registry statistics
   */
  static renderStats(stats: RegistryStats): string {
    return renderToString(<RegistryStatsComponent stats={stats} />);
  }

  /**
   * Render agent comparison
   */
  static renderComparison(
    agent1: OpenComposerAgent,
    agent2: OpenComposerAgent,
  ): string {
    return renderToString(<AgentComparison agent1={agent1} agent2={agent2} />);
  }
}
