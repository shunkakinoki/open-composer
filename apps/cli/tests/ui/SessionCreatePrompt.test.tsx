import { describe, expect, mock, test } from "bun:test";

import { SessionCreatePrompt } from "../../src/components/SessionCreatePrompt.js";
import { render } from "../utils.js";

// Mock the SessionsCli to avoid database operations during testing
mock.module("../../src/services/sessions-cli.js", () => ({
  SessionsCli: {
    createInteractive: mock(async () => 123),
  },
}));

describe("SessionCreatePrompt", () => {
  test("renders name input step when no props provided", () => {
    const mockOnComplete = mock(() => {});
    const mockOnCancel = mock(() => {});

    const { lastFrame } = render(
      <SessionCreatePrompt
        onComplete={mockOnComplete}
        onCancel={mockOnCancel}
      />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });

  test("renders workspace choice step when name is provided", () => {
    const mockOnComplete = mock(() => {});
    const mockOnCancel = mock(() => {});

    const { lastFrame } = render(
      <SessionCreatePrompt
        initialName="Test Session"
        onComplete={mockOnComplete}
        onCancel={mockOnCancel}
      />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });

  test("renders workspace path step when name and choice are provided", () => {
    const mockOnComplete = mock(() => {});
    const mockOnCancel = mock(() => {});

    const { lastFrame } = render(
      <SessionCreatePrompt
        initialName="Test Session"
        initialWorkspaceChoice="existing"
        onComplete={mockOnComplete}
        onCancel={mockOnCancel}
      />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });

  test("renders confirmation step when all props are provided", () => {
    const mockOnComplete = mock(() => {});
    const mockOnCancel = mock(() => {});

    const { lastFrame } = render(
      <SessionCreatePrompt
        initialName="Test Session"
        initialWorkspaceChoice="none"
        onComplete={mockOnComplete}
        onCancel={mockOnCancel}
      />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });
});
