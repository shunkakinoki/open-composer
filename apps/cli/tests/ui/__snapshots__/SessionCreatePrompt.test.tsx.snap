// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`RunCreatePrompt renders name input step when no props provided 1`] = `
"

  🎯 Create New Session

  Enter session name (press Enter for auto-generated):

  _

  Press Enter to continue, Esc to cancel

"
`;

exports[`RunCreatePrompt renders workspace choice step when name is provided 1`] = `
"

  📁 Choose Workspace Option

  Choose workspace option:

  ● 1. Use existing git workspace

  ○ 2. Create new workspace

  ○ 3. No workspace (just session tracking)


  Use ↑↓ to select, Enter to confirm, Esc to cancel

"
`;

exports[`RunCreatePrompt renders workspace path step when name and choice are provided 1`] = `
"

  📂 Select Existing Workspace

  Enter path to existing git workspace:

  _

  Press Enter to continue, Esc to cancel

"
`;

exports[`RunCreatePrompt renders confirmation step when all props are provided 1`] = `
"

  ✅ Confirm Session Creation

  Session Name: Test Session

  Workspace: No workspace assigned


  Press Enter to create session, Esc to cancel

"
`;
